sycl/test/invoke_simd/invoke_simd.cpp

```diff
@@ -64,6 +65,26 @@ class ESIMDSelector : public device_selector {
     return device.is_gpu() ? 1000 : -1;
   }
 };
+#else
+class ESIMDSelector : public device_selector {
+  // Require GPU device
+  virtual int operator()(const device &device) const {
+    if (const char *dev_selector = getenv("ONEAPI_DEVICE_SELECTOR")) {
+      std::string selector_string(dev_selector);
+      if (selector_string.find("gpu") != std::string::npos)
+        return device.is_gpu() ? 1000 : -1;
+      if (selector_string.find("host") != std::string::npos)
+        return device.is_host() ? 1000 : -1;
+      std::cerr
+          << "Supported 'ONEAPI_DEVICE_SELECTOR' env var values are 'gpu', "
+          << selector_string << "' does not contain such substrings.\n";
+      return -1;
+    }
+    // If "ONEAPI_DEVICE_SELECTOR" not defined, only allow gpu device
+    return device.is_gpu() ? 1000 : -1;
+  }
+};
+#endif //__INTEL_PREVIEW_BREAKING_CHANGES
```

Comment:
`ONEAPI_DEVICE_SELECTOR` is available with or without breaking changes. Is there a reason we wouldn't just use it here instead of the deprecated one? In fact, we don't built this test with `-fpreview-breaking-changes` so the new branch is dead code.

In reality I don't even think it matters much. This test is compile-only so this selector really has no effect on whether the test passes or fails.